Automattic/woocommerce-payments
# Move email title hook to the card gateway

Fixes https://github.com/Automattic/woocommerce-payments/issues/6731

#### Changes proposed in this Pull Request
This PR moves the callback's attachment for `woocommerce_email_before_order_table` hook from the UPE class to the parent legacy gateway class. This was done because the legacy gateway class allows registering hooks for the card gateway only. The problem we're trying to fix appears because previously for the split UPE, 

```php
add_action( 'woocommerce_email_before_order_table', [ $this, 'set_payment_method_title_for_email' ], 10, 3 );
```

was added N times where N = number of available UPE gateways, which led to a visible performance downgrade. The hook should be initialized only once and thus was moved to the parent class.









#### Testing instructions





Perform the below instructions sequentially for: **legacy card, legacy UPE, split UPE, deferred intent creation UPE**.

1. Perform steps 2-5 from the testing instructions [here](https://github.com/Automattic/woocommerce-payments/pull/6351) and confirm they are successful
2. Open `access.log` on your server and confirm that the `GET /wp-json/wpcom/v2/sites/${xyz}/wcpay/payment_methods` request is sent only twice and not more.

-------------------


**Post merge**
